src/types.js

```diff
@@ -0,0 +1,71 @@
+var bigi = require('bigi')
+var ecurve = require('ecurve')
+var typeforce = require('typeforce')
+
+function nBuffer (value, n) {
+  if (!Buffer.isBuffer(value)) return false
+  if (value.length !== n) throw new Error('Expected ' + (n * 8) + '-bit Buffer, got ' + (value.length * 8) + '-bit')
+  return true
+}
+
+function Hash160bit (value) { return nBuffer(value, 20) }
+function Hash256bit (value) { return nBuffer(value, 32) }
+function Buffer256bit (value) { return nBuffer(value, 32) }
+
+var UINT53_MAX = Math.pow(2, 53) - 1
+function UInt2 (value) { return (value & 3) === value }
+function UInt8 (value) { return (value & 0xff) === value }
+function UInt32 (value) { return (value >>> 0) === value }
+function UInt53 (value) {
+  return typeforce.Number(value) &&
+    value >= 0 &&
+    value <= UINT53_MAX &&
+    Math.floor(value) === value
+}
+
+// external dependent types
+function BigInt (value) { return !typeforce.Null(value) && value.constructor === bigi }
+function ECCurve (value) { return !typeforce.Null(value) && value.constructor === ecurve.Curve }
+function ECPoint (value) { return !typeforce.Null(value) && value.constructor === ecurve.Point }
```

Comment:
@weilu you this does revert our duck typing changes.   I've never had that problem,  so I'm not sure why we allowed it in the first place.

`ECPair` and `ecdsa` are the only interfaces that accept `bigi` / `ECPoint`'s,  and we [intend](https://github.com/bitcoinjs/bitcoinjs-lib/issues/407) to [remove](https://github.com/bitcoinjs/bitcoinjs-lib/issues/394) their individual use entirely for `2.0.0`,  so this shouldn't even be an issue.


Various bugs where internal versions of these libraries could be inconsistent means duck typing is a very bad idea in this case.